File: LeetCode/Kotlin/s1863.kt
class Solution1863 {
    fun subsetXORSum(nums: IntArray): Int =
        (1..1.shl(nums.size)).fold(0) { r, i ->
            nums.foldIndexed(0) { j, x, n -> if (i.and(1.shl(j)) != 0) x.xor(n) else x } + r
        }
}

File: LeetCode/Rust/s1863.rs
struct Solution1863 {}

impl Solution1863 {
    pub fn subset_xor_sum(nums: Vec<i32>) -> i32 {
        (1..= 1<<nums.len()).fold(0,|r, i|{
            nums.iter().enumerate().fold(0, |x, (j,n)|{ if i & 1 << j == 0 {x} else {x^n} }) + r
        })
    }

}